### PR TITLE
doc generate plugins page from awesome MapLibre

### LINKED ISSUE #4682: Docs: Split new category from UI plugins
## User Story

As a developer using MapLibre I can find the plugin I need more easily in https://maplibre.org/maplibre-gl-js/docs/plugins/ 

## Rationale

[User Interface Plugins](https://maplibre.org/maplibre-gl-js/docs/plugins/#user-interface-plugins) has 24 items and more coming. 

## Impact

It's hard to find the plugin I need

## Solution:
Have more categories.
One new category could be called Search or Geocoding - there are several such plugins in the list
If you have some more ideas for another new category, let me know

### PR DESCRIPTION
**Pull request by lhapaipai**
This PR allows to generate the `plugins.md` documentation page from the contents of the Awesome MapLibre repository

it follows this PR opened in the awesome MapLibre repository
[PR awesome-maplibre/Sync with maplibre doc](https://github.com/maplibre/awesome-maplibre/pull/57)

after discussion it was decided that the maplibre-gl-js documentation page would rely on awesome's links to build the page.

Resolves #4682

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

I not added the `<!-- Do not edit. This file is generated from Awesome MapLibre -->` banner on top of `plugins.md` because the file is already in the gitignore.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -109,6 +109,7 @@
     "mock-geolocation": "^1.0.11",
     "monocart-coverage-reports": "^2.10.4",
     "nise": "^6.1.0",
+    "node-fetch": "^3.3.2",
```
Comment (HarelM) [inline package.json:112]: Is it possible to use node built in http package?

File: build/generate-docs.ts [hunk 2]
```diff
@@ -110,11 +111,82 @@ function generateExamplesFolder() {
     fs.writeFileSync(path.join(examplesDocsFolder, 'index.md'), indexMarkdown);
 }
 
+/**
+ * It extract some content from a reference file (README.md) and replace extracts into
```
Comment (HarelM) [inline build/generate-docs.ts:115]: Is this comment still up-to-date? I don't see a template file...?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,11 @@
 ## main
 
 ### ✨ Features and improvements
-- _...Add new stuff here..._
+
+- Doc plugins page generated from Awesome MapLibre ([#4707](https://github.com/maplibre/maplibre-gl-js/pull/4707))
```
Comment (HarelM) [inline CHANGELOG.md:5]: This isn't interesting enough for changelog, can you remove it please?

**Comment by HarelM** [comment]
Overall looks nice and simply, thanks!

**Comment by lhapaipai** [comment]
Hi @HarelM,
requested changes are done.

last little detail, I wonder if we should add an html comment in the header of the `README.md` of Awesome MapLibre to this file https://github.com/maplibre/maplibre-gl-js/blob/main/build/generate-docs.ts so that people who wonder what these symbols `<!-- [ID]:BEGIN -->` mean have the explanation ?

Have a good day

File: build/generate-docs.ts [hunk 4]
```diff
@@ -110,11 +111,95 @@ function generateExamplesFolder() {
...
+     * ```
+     *    Template header
+     *    CONTENT-TO-EXTRACT
+     * ```
+
+    */
+    const awesomeReadmeUrl = 'https://raw.githubusercontent.com/maplibre/awesome-maplibre/main/README.md';
+    const awesomeReadme = await fetchUrlContent(awesomeReadmeUrl);
+
+    let pluginsTemplate = `# Plugins
```
[the remark this answers is not in the document]
Comment (lhapaipai) [reply build/generate-docs.ts:166]: everything is debatable, it was my choice for more flexibility, if we want to add an header text, put text between the 2 extractions. the objective was to have the rendering that was closest to the existing `plugins.md` page and it was necessary to give a title `## Framework Integrations` for the javascript bindings.

[the remark this answers is not in the document]
Comment (HarelM) [reply build/generate-docs.ts:166]: I don't object to the layout, it is just a bit harder to read due to the flexibility, which I think is an overkill in this case.
I would prefer if the code was a bit more procedural, like the other parts of this file - i.e.
```ts
let pluginContent = "# Plugins\n";
pluginContent += <first part goes here> + "\n\n";
pluginContent += "## Framework Integrations\n\n";
pluginContent += <second part goes here> + "\n\n";
```

Or something similar...

**Review by HarelM — APPROVED** [review]
THANKS!

**Comment by HarelM** [comment]
I've fixed the "respository" typo in awesome maplibre, can you add `plotly` and `topojson` to the cspell config to pass the spell check?

**Pull request merged**